### PR TITLE
+act #17392 Include generated japi.function from akka-stream

### PR DESCRIPTION
**Pull request by patriknw**
- add boilerplate plugin
- make them Serializable to be able to grab line number for Java 8 lambdas

### CODE REVIEW

**Comment by ktoso** [comment]
PLS BUILD

**Comment by patriknw** [comment]
This is for 2.3.11, and I would like @rkuhn to review it

**Comment by rkuhn** [comment]
Good, then, that I’m back on the grid :-)

File: akka-actor/src/main/scala/akka/japi/function/Function.scala [hunk 1]
```diff
@@ -5,22 +5,27 @@ package akka.japi.function
...
-trait Function2[-T1, -T2, +R] {
+@SerialVersionUID(1L)
+trait Function2[-T1, -T2, +R] extends java.io.Serializable {
   @throws(classOf[Exception])
   def apply(arg1: T1, arg2: T2): R
 }
 
 /**
  * A Procedure is like a Function, but it doesn't produce a return value.
+ * `Serializable` is needed to be able to grab line number for Java 8 lambdas.
```
Comment (rkuhn) [inline akka-actor/src/main/scala/akka/japi/function/Function.scala]: but the `Serializable` is missing?

File: akka-actor/src/main/boilerplate/akka/japi/function/Functions.scala.template [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+ * Copyright (C) 2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.japi.function
+
+[3..22#/**
+ * A Function interface. Used to create 1-arg first-class-functions is Java.
+ * `Serializable` is needed to be able to grab line number for Java 8 lambdas.
+ */
+@SerialVersionUID(##1L)
+trait Function1[[#T1#], R] extends java.io.Serializable {
```
Comment (rkuhn) [inline akka-actor/src/main/boilerplate/akka/japi/function/Functions.scala.template]: shouldn’t they be contra-/co-variant?

Comment (rkuhn) [inline akka-actor/src/main/boilerplate/akka/japi/function/Functions.scala.template]: (I think that was my mistake back then, but nevertheless it should be fixed)

File: akka-actor/src/main/scala/akka/japi/function/Function.scala (hunk 1 above)
Comment (patriknw) [inline akka-actor/src/main/scala/akka/japi/function/Function.scala]: ah, good catch

File: akka-actor/src/main/boilerplate/akka/japi/function/Functions.scala.template (hunk 2 above)
Comment (patriknw) [inline akka-actor/src/main/boilerplate/akka/japi/function/Functions.scala.template]: right, now they look like:

    /**
     * A Function interface. Used to create 3-arg first-class-functions is Java.
     * `Serializable` is needed to be able to grab line number for Java 8 lambdas.
     */
    @SerialVersionUID(1L)
    trait Function3[-T1, -T2, -T3, +R] extends java.io.Serializable {
      @throws(classOf[Exception])
      def apply(arg1: T1, arg2: T2, arg3: T3): R
    }
    
    /**
     * A Function interface. Used to create 4-arg first-class-functions is Java.
     * `Serializable` is needed to be able to grab line number for Java 8 lambdas.
     */
    @SerialVersionUID(1L)
    trait Function4[-T1, -T2, -T3, -T4, +R] extends java.io.Serializable {
      @throws(classOf[Exception])
      def apply(arg1: T1, arg2: T2, arg3: T3, arg4: T4): R
    }

    /**
     * A Consumer interface. Used to create 2-arg consumers in Java.
     * A Procedure is like a Function, but it doesn't produce a return value.
     * `Serializable` is needed to be able to grab line number for Java 8 lambdas.
     */
    @SerialVersionUID(1L)
    trait Procedure2[-T1, -T2] extends java.io.Serializable {
      @throws(classOf[Exception])
      def apply(arg1: T1, arg2: T2): Unit
    }
    
    /**
     * A Consumer interface. Used to create 3-arg consumers in Java.
     * A Procedure is like a Function, but it doesn't produce a return value.
     * `Serializable` is needed to be able to grab line number for Java 8 lambdas.
     */
    @SerialVersionUID(1L)
    trait Procedure3[-T1, -T2, -T3] extends java.io.Serializable {
      @throws(classOf[Exception])
      def apply(arg1: T1, arg2: T2, arg3: T3): Unit
    }

**Comment by patriknw** [comment]
updated

File: akka-actor/src/main/boilerplate/akka/japi/function/Functions.scala.template (hunk 2 above)
Comment (ktoso) [inline akka-actor/src/main/boilerplate/akka/japi/function/Functions.scala.template]: Good catch. Yes variance annotations looks right on the new ones.

**Comment by rkuhn** [comment]
LGTM

**Pull request merged**